stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/externimpl/consumer/CreateConsumer.java

```diff
@@ -0,0 +1,194 @@
+ * under the License.
+ *
+ */
+
+package org.ballerinalang.messaging.artemis.externimpl.consumer;
+
+import org.apache.activemq.artemis.api.core.ActiveMQException;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.api.core.client.ClientConsumer;
+import org.apache.activemq.artemis.api.core.client.ClientMessage;
+import org.apache.activemq.artemis.api.core.client.ClientSession;
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BLangVMErrors;
+import org.ballerinalang.bre.bvm.BlockingNativeCallableUnit;
+import org.ballerinalang.bre.bvm.CallableUnitCallback;
+import org.ballerinalang.connector.api.Annotation;
+import org.ballerinalang.connector.api.BLangConnectorSPIUtil;
+import org.ballerinalang.connector.api.BallerinaConnectorException;
+import org.ballerinalang.connector.api.Executor;
+import org.ballerinalang.connector.api.Resource;
+import org.ballerinalang.connector.api.Service;
+import org.ballerinalang.connector.api.Struct;
+import org.ballerinalang.connector.api.Value;
+import org.ballerinalang.messaging.artemis.ArtemisConstants;
+import org.ballerinalang.messaging.artemis.ArtemisUtils;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BError;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.services.ErrorHandlerUtils;
+import org.ballerinalang.util.codegen.ProgramFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Extern function to start the Artemis consumer.
+ *
+ * @since 0.995
+ */
+
+@BallerinaFunction(
+        orgName = ArtemisConstants.BALLERINA, packageName = ArtemisConstants.ARTEMIS,
+        functionName = "createConsumer",
+        receiver = @Receiver(type = TypeKind.OBJECT, structType = ArtemisConstants.LISTENER_OBJ,
+                             structPackage = ArtemisConstants.PROTOCOL_PACKAGE_ARTEMIS)
+)
+public class CreateConsumer extends BlockingNativeCallableUnit {
+    private static final Logger logger = LoggerFactory.getLogger(CreateConsumer.class);
+
+    private static final String FILTER = "filter";
+
+    @Override
+    public void execute(Context context) {
+        try {
+            @SuppressWarnings(ArtemisConstants.UNCHECKED)
+            BMap<String, BValue> consumerObj = (BMap<String, BValue>) context.getRefArgument(0);
+
+            ClientSession session = ArtemisUtils.getClientSessionFromBMap(consumerObj);
+
+            Service service = BLangConnectorSPIUtil.getServiceRegistered(context);
+
+            Annotation serviceAnnotation = getServiceConfigAnnotation(service);
+            Struct annotationValue = serviceAnnotation.getValue();
+            boolean autoAck = annotationValue.getBooleanField("autoAck");
```

Comment:
Shall we create constants for these fields as well? Check the other places.

Reply:
I am using these values in only one place. Hence it is redundant to define constants for each of these values. For values that I am using more than once I have created constants as it is the best practice.